File: LeetCode/p35.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {}

static auto __ = []()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return 0;
}();

class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int i = 0;
        for (i; i < nums.size(); i++)
        {
            // The element is bigger than target
            if (nums[i] > target)
            {
                break;
            }
            // Found
            else if (nums[i] == target)
            {
                return i;
            }
        }

        // Not Found
        return i;
    }
};

File: LeetCode/p35.py
from typing import List

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = -1, len(nums)
        while l + 1 < r:
            mid = (l + r) // 2
            if nums[mid] < target: l = mid
            else: r = mid
        return r